virtcontainers/qemu.go

```diff
@@ -1010,6 +1011,12 @@ func (q *qemu) hotplugMemory(memDev *memoryDevice, op operation) error {
 			memDev.sizeMB, currentMemory, q.config.DefaultMemSz)
 	}
 
+	//memory slot 0 is occupied
```

Comment:
I don't understand this comment - can you provide more detail?

Reply:
Is mem0 maybe either the nvdimm or an initial fixed block that can never be plug/unplugged - @devimc , I suspect you may remember?

Reply:
yep, mem0 is occupied by other components like nvdimm or vmfactory share memory

Reply:
Great - @clarecch , can you add that to the comment please then :-) thx!

Reply:
the problem I see with this implementation is that you can update the memory just once since currently we only have 2 slots, @clarecch have you consider to add an option in the configuration file to make the number of slots configurable? , what is the impact (boot time, memory footprint) of each extra slot? ,  In the case of PODs a memory constraint have to be applied, please take a look to this PR https://github.com/kata-containers/runtime/issues/158

Reply:
For initrd w/o vm factory case, slot 0 is available for hotplug. You can proceed with current code for now. But as @WeiZhang555 mentioned, we should track the slot id somewhere so that we actually know which slot are used and which are not. Then we can drop the hardcoded starting point of slot number.

Reply:
@bergwolf I am using QemuState.Slots, which is stored in hypervisor.json to store the slot id if it has changed now. I think if we have use any mempry slot we should update QemuState.Slots, and then store it to hypervisor.json. I will take a look at components  which will occupied the mem0

Reply:
@devimc Yes, I also think it's needed to add an option in the configuration file to make the number of slots configurable. But maybe create a new issue to do this as well as check if the maximum number of memory slots impacts container memory footprint?

Reply:
@devimc @bergwolf How about use qmp command to track slot number at present, and then initialize the QemuState.Slots field before hotplug memory?
Just like this.
(QEMU) query-memory-devices
{
    "return": [
        {
            "data": {
                "slot": 0,
                "node": 0,
                "addr": 4294967296,
                "memdev": "/objects/mem0",
                "id": "nv0",
                "hotpluggable": true,
                "hotplugged": false,
                "size": 536870912
            },
            "type": "dimm"
        }
    ]
}

Reply:
@clarecch sounds good, in that way we can see what slots are available

Reply:
@clarecch Yes, indeed. We can even remove the slot assignment from kata. We only need to make sure every time we use different memory device id to hotplug it. QEMU always plugs a new dimm in the first available slot. We either query qemu for available slots, or rely on qmp hotplug failure to detect unavailable dimm slots.